pkg/test/framework/components/istio/config.go

```diff
@@ -129,6 +129,10 @@ type Config struct {
 	// CustomSidecarInjectorNamespace allows injecting the sidecar from the specified namespace.
 	// if the value is "", use the default sidecar injection instead.
 	CustomSidecarInjectorNamespace string
+
+	// PartialCleanup will only clean up the configurations generate by istioctl. This avoids removing
+	// extra things like the namespace, webhooks, etc. This is useful for multiple control plane
+	PartialCleanup bool
```

Comment:
The name isn't immediately clear.  Could Istio cleanup just figure this out? Maybe by looking at the revision value? Or maybe when it's installed it can detect if another control plane is already installed (e.g. check for the istio-system namespaces) and it just does the right thing?

Reply:
I don't think we have a great way to do this. I guess we can look at the revision value, but that involves a but of machinery to marshal the yaml and read it

Reply:
Ok that's fair. Would there ever be a reason to do a partial cleanup without multiple control planes?  Also, when we install a second control plane, do we skip deploying the webhook, namespace, etc., or do we just overwrite those resource? If the former, maybe we a better name here `DeploySecondaryIstio`. Alternatively, we could change `DeployIstio` into an enumeration (e.g. Primary/Secondary).

Reply:
I am going to try something different actually, and just always do the "partial" cleanup. This will work correct in both cases, and is what we actually document as how to clean up, so probably makes the most sense